src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs

```diff
@@ -37,7 +37,7 @@ internal sealed partial class MethodInvoker
                 }
                 else
                 {
-                    if (RuntimeFeature.IsDynamicCodeCompiled)
+                    if (RuntimeFeature.IsDynamicCodeSupported)
```

Comment:
This is going to affect more than just WASM: Android too, I think.
/cc @ivanpovazan

Reply:
> From the link above it appears Android already has `IsDynamicCodeCompiled == true` since it is using the emit path already. If that is the case, then this PR shouldn't affect it.

> So this PR loosens the requirement to say if dynamic code is supported at all, then have reflection emit generate dynamic code even if it is interpreted.

I  see, makes sense.